src/shared/modules/auth/index.js

```diff
@@ -0,0 +1,242 @@
+import {
+  getCredentialsFromAuthResult,
+  getInitialisationParameters,
+  getSSOProviderByIdpId,
+  temporarlyStoreUrlSearchParams
+} from './common'
+import {
+  AUTH_STORAGE_CODE_VERIFIER,
+  AUTH_STORAGE_STATE,
+  BEARER,
+  IMPLICIT,
+  PKCE
+} from './constants'
+import {
+  createStateForRequest,
+  createNonce,
+  createCodeVerifier,
+  createCodeChallenge,
+  authLog,
+  authDebug,
+  removeSearchParamsInBrowserHistory
+} from './helpers'
+import {
+  defaultCodeChallengeMethod,
+  defaultGrantType,
+  searchParamsToRemoveAfterAuthRedirect
+} from './settings'
+
+export const authRequestForSSO = idpId => {
```

Comment:
The returned error maybe should be an Error, or maybe even thrown? This would make it consistent with other errors caught/used in discoveryDuck